airbyte-integrations/connectors/destination-meilisearch/destination_meilisearch/writer.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import time
+from collections.abc import Mapping
+from uuid import uuid4
+
+from meilisearch import Client
+
+
+class MeiliWriter:
+    write_buffer = []
+    flush_interval = 50000
+
+    def __init__(self, client: Client, steam_name: str, primary_key: str):
+        self.client = client
+        self.steam_name = steam_name
+        self.primary_key = primary_key
+
+    def queue_write_operation(self, data: Mapping):
+        random_key = str(uuid4())
+        self.write_buffer.append({**data, self.primary_key: random_key})
+        if len(self.write_buffer) == self.flush_interval:
+            self.flush()
+
+    def flush(self):
+        if len(self.write_buffer) == 0:
+            return
+
+        response = self.client.index(self.steam_name).add_documents(self.write_buffer)
+        self.wait_for_job(response.task_uid)
+        self.write_buffer.clear()
```

Comment:
Please add a logger message saying you're flushing the records.

Reply:
Added!